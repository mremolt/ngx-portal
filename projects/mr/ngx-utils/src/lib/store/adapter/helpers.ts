import { NormalizedEntities } from './types';

export function mergeNormalizedEntities<Entity extends object>(
  currentEntities: NormalizedEntities<Entity>,
  newEntities: NormalizedEntities<Entity>
): NormalizedEntities<Entity> {
  return Object.keys(newEntities).reduce((result, key) => {
    const keyResult = { ...currentEntities[key], ...newEntities[key] };
    return { ...result, [key]: keyResult };
  }, {});
}

export function removeProps<T extends { [key: string]: any }>(obj: T, propNames: Array<keyof T>) {
  return Object.keys(obj).reduce((result, key: string) => {
    if (propNames.includes(key)) {
      return result;
    } else {
      return { ...result, [key]: obj[key] };
    }
  }, {});
}

export function removeProp<T extends { [key: string]: any }>(obj: T, propName: keyof T) {
  return removeProps(obj, [propName]);
}
