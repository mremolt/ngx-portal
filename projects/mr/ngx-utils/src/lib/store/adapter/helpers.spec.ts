import { removeProp, removeProps } from './helpers';
import { User, user1 } from './test.fixtures';

describe('helpers', () => {
  describe('mergeNormalizedEntities', () => {});

  describe('removeProps', () => {
    let subject: User;

    beforeEach(() => {
      subject = { ...user1 };
    });

    it('removes given properties from the target object', () => {
      expect(removeProps(subject, ['firstname', 'planet'])).toMatchSnapshot();
    });

    it('does not change the original object', () => {
      removeProps(subject, ['planet', 'id', 'lastname']);
      expect(subject).toEqual(user1);
    });
  });

  describe('removeProp', () => {
    let subject: User;

    beforeEach(() => {
      subject = { ...user1 };
    });

    it('removes given property from the target object', () => {
      expect(removeProp(subject, 'firstname')).toMatchSnapshot();
    });

    it('does not change the original object', () => {
      removeProp(subject, 'planet');
      expect(subject).toEqual(user1);
    });
  });
});
