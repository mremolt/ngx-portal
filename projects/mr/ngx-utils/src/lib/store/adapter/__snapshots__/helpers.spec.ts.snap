// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers removeProp removes given property from the target object 1`] = `
Object {
  "hobbies": Array [
    Object {
      "id": "12",
      "name": "Drinking tea",
    },
    Object {
      "id": "13",
      "name": "Carrying towels",
    },
  ],
  "id": "42",
  "lastname": "Dent",
  "planet": Object {
    "id": "43",
    "name": "Earth",
  },
}
`;

exports[`helpers removeProps removes given properties from the target object 1`] = `
Object {
  "hobbies": Array [
    Object {
      "id": "12",
      "name": "Drinking tea",
    },
    Object {
      "id": "13",
      "name": "Carrying towels",
    },
  ],
  "id": "42",
  "lastname": "Dent",
}
`;
